homeassistant/components/viam/services.py

```diff
@@ -0,0 +1,293 @@
+"""Services for Viam integration."""
+
+from __future__ import annotations
+
+from datetime import datetime
+from functools import partial
+
+from viam.app.app_client import RobotPart
+from viam.services.vision import VisionClient
+import voluptuous as vol
+
+from homeassistant.components import camera
+from homeassistant.config_entries import ConfigEntry, ConfigEntryState
+from homeassistant.core import (
+    HomeAssistant,
+    ServiceCall,
+    ServiceResponse,
+    SupportsResponse,
+    callback,
+)
+from homeassistant.exceptions import ServiceValidationError
+from homeassistant.helpers import selector
+
+from .const import (
+    DOMAIN,
+    SERVICE_CAMERA,
+    SERVICE_CLASSIFIER_NAME,
+    SERVICE_COMPONENT_NAME,
+    SERVICE_COMPONENT_TYPE,
+    SERVICE_CONFIDENCE,
+    SERVICE_COUNT,
+    SERVICE_DETECTOR_NAME,
+    SERVICE_FILE_NAME,
+    SERVICE_FILEPATH,
+    SERVICE_ROBOT_ADDRESS,
+    SERVICE_ROBOT_SECRET,
+    SERVICE_VALUES,
+)
+from .manager import ViamManager
+
+ATTR_CONFIG_ENTRY = "config_entry"
+
+DATA_CAPTURE_SERVICE_NAME = "capture_data"
+CAPTURE_IMAGE_SERVICE_NAME = "capture_image"
+CLASSIFICATION_SERVICE_NAME = "get_classifications"
+DETECTIONS_SERVICE_NAME = "get_detections"
+
+ENTRY_SERVICE_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_CONFIG_ENTRY): selector.ConfigEntrySelector(
+            {
+                "integration": DOMAIN,
+            }
+        ),
+    }
+)
+DATA_CAPTURE_SERVICE_SCHEMA = ENTRY_SERVICE_SCHEMA.extend(
```

Comment:
Is capture a good name considering the service does not capture data by itself but accepts data as input?

Reply:
"Data Capture" is a Viam term (https://docs.viam.com/data/capture/) that respond to the `tabular_data_capture_upload` method from the Python SDK used by the respective service. It will be familiar for folks when searching the Viam documentation.